### PR TITLE
Add archive and exportArchive for iOS.

### LINKED ISSUE #1790: tns publish ios fails with cocoapods
Trying the `tns publish` command with the QSF fails with:

```
Code Sign error: Provisioning profile does not match bundle identifier: The provisioning
profile specified in your build settings ("OrgNativeScriptDistribution") has an AppID of
"org.nativescript.*" which does not match your bundle identifier
"org.cocoapods.IQKeyboardManager".
```

I don't think we should try to sign any of the cocoapod targets using the App's provisioning profile.

### PR DESCRIPTION
**Pull request by PanayotCankov**
Resolves #1790

Tests for tns appstore with xcode archives

Added script that runs tsc --watch and upon changes runs mocha, good for unit test writing

Let us unit test easier, no references .d.ts anymore, no use strict anymore (ts will add this automagically)

Fixing unreachable code

Add tests for IOSProjectService archive and exportArchive

Added `npm run mocha` and `npm run tsc` to run `tsc` and `mocha` from dev dependencies.

Added `npm run test-watch` script that runs `tsc --watch` and on successful compilation runs also `mocha` from dev dependencies

### CODE REVIEW

**Comment by enchev** [comment]
:+1:

**Comment by PanayotCankov** [comment]
Fail :(
[We want to submit more than iossampleapp.app.](https://github.com/NativeScript/nativescript-cli/blob/master/lib/services/itmstransporter-service.ts#L149)

File: lib/services/itmstransporter-service.ts [hunk 1]
```diff
@@ -146,7 +124,22 @@ export class ITMSTransporterService implements IITMSTransporterService {
 					temp.track();
 					let destinationDir = temp.mkdirSync("ipa-");
 					this.$fs.unzip(ipaFileFullPath, destinationDir).wait();
-					let plistObject = this.$bplistParser.parseFile(path.join(destinationDir, "Payload", "iossampleapp.app", "Info.plist")).wait();
+					// WHAT THE FUCK! WHAT THE FUCKING FUUUCK!
```
Comment (rosen-vladimirov) [inline lib/services/itmstransporter-service.ts]: hahaha

**Comment by rosen-vladimirov** [comment]
:+1: I love the tests

**Comment by PanayotCankov** [comment]
Aside the hardcoded app name in the ITMSTransporterService it was pleasure working with it.
I am merging this once all builds pass.

**Pull request merged**